src/streamingCalls/streaming.ts

```diff
@@ -119,6 +129,19 @@ export class StreamProxy extends duplexify implements GRPCCallResult {
       stream.on(event, this.emit.bind(this, event));
     });
 
+    // gRPC guaranteed emit the 'status' event but not 'metadata'. And 'status' is the last event to be emit.
+    // Emit the 'response' event if stream has no 'metadata' event.
+    // This avoids stream swallow the other events, such as 'end'.
+    stream.on('status', () => {
+      if (!this._responseHasSent) {
+        stream.emit('response', {
+          code: 200,
```

Comment:
`status` always indicates a `200`?

Reply:
Try to do the same thing in [code](https://github.com/googleapis/gax-nodejs/blame/main/src/streamingCalls/streaming.ts#L131-L136)